libcalico-go/lib/ipam/ipam_upgrade_kdd_test.go

```diff
@@ -0,0 +1,134 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/projectcalico/calico/libcalico-go/lib/apiconfig"
+	libapiv3 "github.com/projectcalico/calico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/calico/libcalico-go/lib/backend"
+	bapi "github.com/projectcalico/calico/libcalico-go/lib/backend/api"
+	"github.com/projectcalico/calico/libcalico-go/lib/backend/k8s"
+	k8sresources "github.com/projectcalico/calico/libcalico-go/lib/backend/k8s/resources"
+	"github.com/projectcalico/calico/libcalico-go/lib/names"
+	cnet "github.com/projectcalico/calico/libcalico-go/lib/net"
+	"github.com/projectcalico/calico/libcalico-go/lib/testutils"
+)
+
+// createUnlabeledBlockAffinity creates a BlockAffinity CRD directly through the CRD REST client
+// without any labels. Name must be host + "-" + CIDRToName to match Calico conventions.
+func createUnlabeledBlockAffinity(ctx context.Context, rc rest.Interface, host string, cidr string) error {
+	_, ipn, err := cnet.ParseCIDR(cidr)
+	if err != nil {
+		return err
+	}
+	name := fmt.Sprintf("%s-%s", host, names.CIDRToName(*ipn))
+	ba := &libapiv3.BlockAffinity{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       libapiv3.KindBlockAffinity,
+			APIVersion: "crd.projectcalico.org/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: libapiv3.BlockAffinitySpec{
+			State:   string(v3.StateConfirmed),
+			Node:    host,
+			Type:    "host",
+			CIDR:    cidr,
+			Deleted: "false",
+		},
+	}
+	return rc.Post().Resource(k8sresources.BlockAffinityResourceName).Body(ba).Do(ctx).Error()
+}
+
+var _ = testutils.E2eDatastoreDescribe("IPAM UpgradeHost (Kubernetes datastore only)", testutils.DatastoreK8s, func(config apiconfig.CalicoAPIConfig) {
+	var (
+		bc         bapi.Client
+		ic         Interface
+		restClient *rest.RESTClient
+	)
+
+	BeforeEach(func() {
+		// Fresh backend client and IPAM client.
+		var err error
+		bc, err = backend.NewClient(config)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(bc.Clean()).NotTo(HaveOccurred())
+		ic = NewIPAMClient(bc, ipPools, &fakeReservations{})
+
+		// Build a raw CRD REST client using the same kubeconfig as the backend client.
+		cfg, _, err := k8s.CreateKubernetesClientset(&config.Spec)
+		Expect(err).NotTo(HaveOccurred())
+		rc, err := k8s.RawCRDClientV1(*cfg)
+		Expect(err).NotTo(HaveOccurred())
+		restClient = rc
```

Comment:
Is there a reason to create the client as `rc` and then assign it to `restClient` as opposed to just doing `restClient, err = k8s.RawCRDClientV1()` ?